virtcontainers/sandbox.go

```diff
@@ -758,6 +776,13 @@ func createSandbox(ctx context.Context, sandboxConfig SandboxConfig, factory Fac
 		return nil, err
 	}
 
+	defer func() {
+		// when fetchSandbox(), factory is nil, we should not release resources
+		if err != nil && factory != nil {
```

Comment:
I'm not sure I get the reasoning. Except for `fetchSandbox()`, `factory` can still be nil when vm factory is disabled.

Reply:
:+1: important point! Now I move this part after comment `// Below code path is called only during create, because of earlier check.` Is it the right place now? :thinking: